walk-in-interview/src/main/java/com/google/job/data/JobsDatabase.java

```diff
@@ -0,0 +1,72 @@
+package com.google.job.data;
+
+import com.google.api.core.ApiFunction;
+import com.google.api.core.ApiFuture;
+import com.google.api.core.ApiFutures;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.WriteResult;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.firestore.FireStoreUtils;
+import org.checkerframework.checker.nullness.compatqual.NullableDecl;
+
+import java.util.concurrent.Future;
+
+/** Helps persist and retrieve job posts. */
+public final class JobsDatabase {
+    private static final String JOB_COLLECTION = "Jobs";
+
+    private final Firestore firestore;
+
+    public JobsDatabase(Firestore firestore) {
+        this.firestore = firestore;
+    }
+
+    /**
+     * Adds a newly created job post.
+     *
+     * @param newJob Newly created job post. Assumes that it is non-nullable.
+     * @return Future for the added job post task, convenient for testing.
+     */
+    public Future<DocumentReference> addJob(Job newJob) {
+        Future<DocumentReference> addedDocRef = firestore.collection(JOB_COLLECTION).add(newJob);
+        return addedDocRef;
+    }
+
+    /**
+     * Edits the job post.
+     *
+     * @param jobId Id for the target job post in the database.
+     * @param job Updated job post.
+     * @return A future of the detailed information of the update.
+     */
+    public Future<WriteResult> editJob(String jobId, Job job) {
+        // Overwrites the whole job post
+        Future<WriteResult> edittedDocRef = firestore.collection(JOB_COLLECTION).document(jobId).set(job);
```

Comment:
I added a check to check if the document(jobId) exist or not

in terms of "edit job post not belong to current user", this will be implemented later, since in the current stage, account is not involved yet